src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -68,45 +68,18 @@ namespace po = boost::program_options;
 // ------ L O C A L   F U N C T I O N S ---------------------------------------
 
 namespace {
-/*
- * Update shell on the board for auto flash
- */
-static void 
-update_shell(unsigned int index, const std::string& primary, const std::string& secondary)
-{
-  Flasher flasher(index);
-  if(!flasher.isValid())
-    throw xrt_core::error(boost::str(boost::format("%d is an invalid index") % index));
-
-  if (primary.empty())
-    throw xrt_core::error("Shell not specified");
-
-  auto pri = std::make_unique<firmwareImage>(primary.c_str(), MCS_FIRMWARE_PRIMARY);
-  if (pri->fail())
-    throw xrt_core::error(boost::str(boost::format("Failed to read %s") % primary));
-
-  std::unique_ptr<firmwareImage> sec;
-  if (!secondary.empty()) {
-    sec = std::make_unique<firmwareImage>(secondary.c_str(),
-      MCS_FIRMWARE_SECONDARY);
-    if (sec->fail())
-      sec = nullptr;
-  }
-  
-  if (flasher.upgradeFirmware("", pri.get(), sec.get()) != 0)
-    throw xrt_core::error("Failed to update base");
-  
-  std::cout << boost::format("%-8s : %s \n") % "INFO" % "Base flash image has been programmed successfully.";
-}
 
-/*
- * Update shell on the board for manual flash
+/**
+ * @brief Update shell on the board for manual flash
+ * 
+ * @param[in] index The index of the board to be flashed
+ * @param[in] image_paths The images to plash onto the board
+ * @param[in] flash_type How the board should be flashed. Typically left as blank
  */
 static void 
-update_shell(unsigned int index, const std::string& flashType,
-  const std::string& primary, const std::string& secondary)
+update_shell(unsigned int index, const std::vector<std::string>& image_paths, const std::string& flash_type="")
```

Comment:
```suggestion
update_shell(unsigned int index, const std::vector<std::string>& image_paths, const std::string& flash_type = "")
```
Check also the line is shorter than 80 characters.

Reply:
Is 80 lines the max length or 120? There are a lot of statements that go over 80